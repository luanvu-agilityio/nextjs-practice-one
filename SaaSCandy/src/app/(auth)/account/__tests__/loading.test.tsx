import { render } from '@testing-library/react';
import AccountLoading from '@/app/(auth)/account/loading';

describe('AccountLoading - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<AccountLoading />);
    expect(container).toMatchSnapshot();
  });

  it('should render all skeleton elements', () => {
    const { container } = render(<AccountLoading />);
    const skeletons = container.querySelectorAll(
      '.animate-pulse, [class*="Skeleton"]'
    );

    // Verify multiple skeleton elements are present
    expect(skeletons.length).toBeGreaterThan(0);
  });
});
