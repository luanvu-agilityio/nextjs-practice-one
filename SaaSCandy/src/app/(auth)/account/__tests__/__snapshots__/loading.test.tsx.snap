// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AccountLoading - Snapshot Tests should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center max-w-[1296px] mx-auto min-h-screen py-8"
  >
    <div
      class="pb-12 w-full"
    >
      <div
        class="flex justify-center mb-4"
      >
        <div
          class="animate-pulse rounded-md bg-muted h-10 w-48"
        />
      </div>
      <div
        class="flex justify-center gap-2"
      >
        <div
          class="animate-pulse rounded-md bg-muted h-4 w-12"
        />
        <div
          class="animate-pulse rounded-md bg-muted h-4 w-2"
        />
        <div
          class="animate-pulse rounded-md bg-muted h-4 w-16"
        />
      </div>
    </div>
    <div
      class="py-14 px-16 w-full max-w-159 rounded-4xl border border-form-border-color shadow-form bg-white animate-pulse"
    >
      <div
        class="flex flex-col gap-10"
      >
        <div
          class="flex justify-center"
        >
          <div
            class="animate-pulse rounded-md bg-muted h-10 w-48"
          />
        </div>
        <div
          class="space-y-8"
        >
          <div
            class="text-center space-y-4"
          >
            <div
              class="flex justify-center"
            >
              <div
                class="animate-pulse bg-muted w-24 h-24 rounded-full"
              />
            </div>
            <div
              class="space-y-2"
            >
              <div
                class="animate-pulse rounded-md bg-muted h-6 w-32 mx-auto"
              />
              <div
                class="animate-pulse rounded-md bg-muted h-4 w-48 mx-auto"
              />
            </div>
          </div>
          <div
            class="space-y-4"
          >
            <div
              class="bg-blue-foreground opacity-80 rounded-lg p-6"
            >
              <div
                class="animate-pulse rounded-md bg-muted h-6 w-40 mb-4"
              />
              <div
                class="space-y-3"
              >
                <div
                  class="flex justify-between items-center"
                >
                  <div
                    class="animate-pulse rounded-md bg-muted h-5 w-12"
                  />
                  <div
                    class="animate-pulse rounded-md bg-muted h-5 w-32"
                  />
                </div>
                <div
                  class="flex justify-between items-center"
                >
                  <div
                    class="animate-pulse rounded-md bg-muted h-5 w-12"
                  />
                  <div
                    class="animate-pulse rounded-md bg-muted h-5 w-48"
                  />
                </div>
                <div
                  class="flex justify-between items-center"
                >
                  <div
                    class="animate-pulse rounded-md bg-muted h-5 w-24"
                  />
                  <div
                    class="animate-pulse bg-muted h-6 w-28 rounded-full"
                  />
                </div>
              </div>
            </div>
            <div
              class="bg-blue-foreground opacity-80 rounded-lg p-6"
            >
              <div
                class="animate-pulse rounded-md bg-muted h-6 w-32 mb-4"
              />
              <div
                class="space-y-3"
              >
                <div
                  class="animate-pulse bg-muted h-12 w-full rounded-lg"
                />
                <div
                  class="animate-pulse bg-muted h-12 w-full rounded-lg"
                />
                <div
                  class="animate-pulse bg-muted h-12 w-full rounded-lg"
                />
              </div>
            </div>
          </div>
          <div
            class="flex gap-4 pt-6"
          >
            <div
              class="animate-pulse bg-muted h-12 flex-1 rounded-lg"
            />
            <div
              class="animate-pulse bg-muted h-12 flex-1 rounded-lg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
